app/main.py

```diff
@@ -685,13 +688,14 @@ def build_filetypes_table(osparc_viewers):
 
 
 @app.route("/sim/dataset/<id_>")
-def sim_dataset(id_):
+@app.route("/sim/dataset/<id_>/<debug_>")
```

Comment:
I don't like having a route for debugging, BUT that warning (and also error further down the code) is only relevant for SIM-Core based simulation datasets (depending on what you do with the `/sim/dataset/<id_>` endpoint), not CellML-based simulation datasets.

Here, I am trying to reuse the `/sim/dataset/<id_>` endpoint and that warning (and error) is not relevant to me, no matter whether it's a SIM-Core based simulation or a CellML-based one. So, I just don't want to see that message at all, be it in production or in my development environment. It clutters my console for no good reason.

I could have copied/pasted most of that code, but that would have been a bit silly, hence the debugging route approach.

Reply:
The debug flag could be setup using a environment variable instead. The debug route is not useful to the person calling it unless they are also running the server.

Reply:
> A debug flag could be setup using a environment variable instead. The debug route is not useful to the person calling it unless they are also running the server.

Sure, although such an environment variable would normally be set IF we want to get debug information while, here, debug information is provided by default (which is wrong in the first place).

Anyway, what should such an environment variable be called? What about `SPARC_API_DEBUGGING`? If it's not set or set to `1` (or `ON`?) then debugging would be on, and if it is set to `0` (or `OFF`?) then debugging would be off.

Reply:
SPARC_API_DEBUGGING is good. Having it default to be on is consistent with the DEPLOY_ENV = development so that is fine. TRUE or FALSE is probably more consistent with standard although it is going to be string comparison regardless.

Reply:
> SPARC_API_DEBUGGING is good. Having it default to be on is consistent with the DEPLOY_ENV = development so that is fine. TRUE or FALSE is probably more consistent with standard although it is going to be string comparison regardless.

Ok, going to use `SPARC_API_DEBUGGING` with unset meaning `TRUE`.